#using try and except

#without using exception handling
def spam(divideBy):
        return 42 / divideBy
        print('invalid argument.')

print(spam(2))
print(spam(12))
print(spam(0))
print(spam(1))


#using exception handling
def spam(divideBy):
    try:
        return 42 / divideBy
    except ZeroDivisionError:
        print('invalid argument.')

print(spam(2))
print(spam(12))
print(spam(0))
print(spam(1))

#alternative way of exception handling
def spam(divideBy):
    return 42 / divideBy

try:
    print(spam(2))
    print(spam(12))
    print(spam(0))
    print(spam(1))
    
except ZeroDivisionError:
    print('Error: Invalid argument.')
